
This is an attempt to get a biarch compiler without full multilib stuff.

The 2nd chunk is needed to not embedded a lib64/ld-linus.so into the
executable.

I _know_ this is all a bit hacky - but migrating all the packages and scripts
to a lib/lib64 seperation is a huge efford, without much real gain ...

  - Rene Rebe <rene@exactcode.de>

--- ./gcc/config/i386/t-linux64.orig	2005-01-17 19:15:04.986820056 +0100
+++ ./gcc/config/i386/t-linux64	2005-01-17 19:15:46.038579240 +0100
@@ -4,9 +4,9 @@
 SHLIB_MAPFILES = $(srcdir)/libgcc-std.ver \
 		 $(srcdir)/config/i386/libgcc-x86_64-glibc.ver
 
-MULTILIB_OPTIONS = m64/m32
-MULTILIB_DIRNAMES = 64 32 
-MULTILIB_OSDIRNAMES = ../lib64 ../lib
+MULTILIB_OPTIONS = m64
+MULTILIB_DIRNAMES = 64
+MULTILIB_OSDIRNAMES = ../lib
 
 LIBGCC = stmp-multilib
 INSTALL_LIBGCC = install-multilib
--- ./gcc/config/sparc/t-linux64.orig	2005-01-17 19:20:21.991628000 +0100
+++ ./gcc/config/sparc/t-linux64	2005-01-17 19:22:44.576951736 +0100
@@ -1,6 +1,6 @@
-MULTILIB_OPTIONS = m64/m32
-MULTILIB_DIRNAMES = 64 32
-MULTILIB_OSDIRNAMES = ../lib64 ../lib
+MULTILIB_OPTIONS = m64
+MULTILIB_DIRNAMES = 64
+MULTILIB_OSDIRNAMES = ../lib
 
 LIBGCC = stmp-multilib
 INSTALL_LIBGCC = install-multilib
--- ./gcc/config/i386/linux64.h.orig	2005-01-19 01:10:35.592295784 +0100
+++ ./gcc/config/i386/linux64.h	2005-01-19 01:11:25.247747008 +0100
@@ -61,7 +61,7 @@
     %{!static: \
       %{rdynamic:-export-dynamic} \
       %{m32:%{!dynamic-linker:-dynamic-linker /lib/ld-linux.so.2}} \
-      %{!m32:%{!dynamic-linker:-dynamic-linker /lib64/ld-linux-x86-64.so.2}}} \
+      %{!m32:%{!dynamic-linker:-dynamic-linker /lib/ld-linux.so.2}}} \
     %{static:-static}}"
 
 #define MULTILIB_DEFAULTS { "m64" }









--- ./gcc/config/sparc/linux64.h.orig	2005-01-19 01:07:56.434491448 +0100
+++ ./gcc/config/sparc/linux64.h	2005-01-19 01:10:09.928197320 +0100
@@ -176,12 +176,12 @@
         %{static:-static}}} \
 "
 
-#define LINK_ARCH64_SPEC "-m elf64_sparc -Y P,/usr/lib64 %{shared:-shared} \
+#define LINK_ARCH64_SPEC "-m elf64_sparc -Y P,/usr/lib %{shared:-shared} \
   %{!shared: \
     %{!ibcs: \
       %{!static: \
         %{rdynamic:-export-dynamic} \
-        %{!dynamic-linker:-dynamic-linker /lib64/ld-linux.so.2}} \
+        %{!dynamic-linker:-dynamic-linker /lib/ld-linux.so.2}} \
         %{static:-static}}} \
 "
 
@@ -257,12 +257,12 @@
 #else /* !SPARC_BI_ARCH */
 
 #undef LINK_SPEC
-#define LINK_SPEC "-m elf64_sparc -Y P,/usr/lib64 %{shared:-shared} \
+#define LINK_SPEC "-m elf64_sparc -Y P,/usr/lib %{shared:-shared} \
   %{!shared: \
     %{!ibcs: \
       %{!static: \
         %{rdynamic:-export-dynamic} \
-        %{!dynamic-linker:-dynamic-linker /lib64/ld-linux.so.2}} \
+        %{!dynamic-linker:-dynamic-linker /lib/ld-linux.so.2}} \
         %{static:-static}}} \
 %{mlittle-endian:-EL} \
 %{!mno-relax:%{!r:-relax}} \
